test/fixtures/circular/array-in-object.json

```diff
@@ -126,7 +126,7 @@
                                   "content": "application/json"
                                 }
                               },
-                              "content": "{\n  \"id\": \"1\",\n  \"data\": []\n}"
+                              "content": "{\n  \"id\": \"1\",\n  \"data\": [\n    {}\n  ]\n}"
```

Comment:
In this case, I do not think the object should be rendered in the array. As the object can have required values that wouldn't be rendered. In that case the new renderer can produce JSON examples that wouldn't pass validation under the generated JSON Schema.

This can be demonstrated if this circular test fixture was updated to have a required parameter (and use fixed-type) on the array:

```apib
# Group Example
# GET /
+ Response 200 (application/json)
    + Attributes (A)

# Data Structures

# A (object)
- id: 1 (required)
- data (array[A], fixed-type)
```

Which would produce the following:

```json
{
  "id": "1",
  "data": [
    {}
  ]
}
```

Unfortunately, it seems the JSON Schema isn't generated as I expected and thus the example I was trying to share doesn't work. As it doesn't use `definitions` to support circular schema generation (I though that was added in recent changes).

```json
{
  "$schema": "http://json-schema.org/draft-04/schema#",
  "type": "object",
  "properties": {
    "id": {
      "type": "string"
    },
    "data": {
      "type": "array",
      "items": {
        "anyOf": [
          {
            "type": "object"
          }
        ]
      }
    }
  },
  "required": [
    "id"
  ]
}
```